dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -2979,7 +2985,12 @@ Spdp::SpdpTransport::handle_input(ACE_HANDLE h)
         DCPS::RcHandle<Spdp> outer = outer_.lock();
 
         if (outer) {
-          outer->data_received(data, plist, remote);
+          if (valid_size(remote)) {
```

Comment:
Move this up as far as it can go -- in other words, why not check `remote`'s validity right after that object is written to?